rust/worker/src/blockstore/arrow/blockfile.rs

```diff
@@ -107,6 +111,49 @@ impl ArrowBlockfileWriter {
         Ok(flusher)
     }
 
+    pub(crate) fn split_and_add<K: ArrowWriteableKey, V: ArrowWriteableValue>(
+        &self,
+        prefix: &str,
+        key: K,
+        value: V,
+        delta: BlockDelta,
+    ) -> Result<(), Box<dyn ChromaError>> {
+        let search_key = CompositeKey::new(prefix.to_string(), key.clone());
+        if delta.can_add(prefix, &key, &value) {
+            delta.add(prefix, key, value);
+        } else {
+            let (split_key, new_delta) = delta.split::<K, V>();
+            self.sparse_index.add_block(split_key, new_delta.id);
+            match new_delta.get_min_key() {
```

Comment:
somewhat prefer that we always leave the job of deciding "where to put the key" to the function used everywhere which is sparse_index.get_target_block_id(). This localizes the splitting logic in a way where inconsistencies are handled.

Reply:
i think this is fine